amd_openvx_extensions/amd_rpp/source/Brightness.cpp

```diff
@@ -152,209 +152,85 @@ static vx_status VX_CALLBACK validateBrightness(vx_node node, const vx_reference
-        data->src_desc_ptr->strides.hStride = data->src_desc_ptr->c * data->src_desc_ptr->w;
-        data->src_desc_ptr->strides.wStride = data->src_desc_ptr->c;
-        data->src_desc_ptr->strides.cStride = 1;
-        data->src_desc_ptr->layout = RpptLayout::NHWC;
-
-        //destination_description_ptr
-        data->dst_desc_ptr->n = data->out_tensor_dims[0] * data->in_tensor_dims[1];
-        data->dst_desc_ptr->h = data->out_tensor_dims[1];
-        data->dst_desc_ptr->w = data->out_tensor_dims[2];
-        data->dst_desc_ptr->c = data->out_tensor_dims[3];
-        data->dst_desc_ptr->strides.nStride = data->dst_desc_ptr->c * data->dst_desc_ptr->w * data->dst_desc_ptr->h;
-        data->dst_desc_ptr->strides.hStride = data->dst_desc_ptr->c * data->dst_desc_ptr->w;
-        data->dst_desc_ptr->strides.wStride = data->dst_desc_ptr->c;
-        data->dst_desc_ptr->strides.cStride = 1;
-        data->dst_desc_ptr->layout = RpptLayout::NHWC;
-    }
-    else if(data->layout == 3) // NFCHW
-    {
-        // source_description_ptr
-        data->src_desc_ptr->n = data->in_tensor_dims[0] * data->in_tensor_dims[1];
-        data->src_desc_ptr->h = data->in_tensor_dims[3];
-        data->src_desc_ptr->w = data->in_tensor_dims[4];
-        data->src_desc_ptr->c = data->in_tensor_dims[2];
-        data->src_desc_ptr->strides.nStride = data->src_desc_ptr->c * data->src_desc_ptr->w * data->src_desc_ptr->h;
-        data->src_desc_ptr->strides.cStride = data->src_desc_ptr->w * data->src_desc_ptr->h;
-        data->src_desc_ptr->strides.hStride = data->src_desc_ptr->w;
-        data->src_desc_ptr->strides.wStride = 1;
-        data->src_desc_ptr->layout = RpptLayout::NCHW;
-
-        //destination_description_ptr
-        data->dst_desc_ptr->n = data->out_tensor_dims[0] * data->in_tensor_dims[1];
-        data->dst_desc_ptr->h = data->out_tensor_dims[1];
-        data->dst_desc_ptr->w = data->out_tensor_dims[2];
-        data->dst_desc_ptr->c = data->out_tensor_dims[3];
-        data->dst_desc_ptr->strides.nStride = data->dst_desc_ptr->c * data->dst_desc_ptr->w * data->dst_desc_ptr->h;
-        data->dst_desc_ptr->strides.hStride = data->dst_desc_ptr->c * data->dst_desc_ptr->w;
-        data->dst_desc_ptr->strides.wStride = data->dst_desc_ptr->c;
-        data->dst_desc_ptr->strides.cStride = 1;
-        data->dst_desc_ptr->layout = RpptLayout::NCHW; 
-    }
-
-#if ENABLE_HIP
-    if (data->device_type == AGO_TARGET_AFFINITY_GPU)
-        hipMalloc(&data->roi_tensor_ptr_dev, data->src_desc_ptr->n * sizeof(RpptROI));
-#endif
-    data->roi_tensor_ptr = (RpptROI *)calloc(data->src_desc_ptr->n, sizeof(RpptROI));
-    data->alpha = (vx_float32 *)malloc(sizeof(vx_float32) * data->src_desc_ptr->n);
-    data->beta = (vx_float32 *)malloc(sizeof(vx_float32) * data->src_desc_ptr->n);
+    data->alpha = (vx_float32 *)malloc(sizeof(vx_float32) * data->srcDescPtr->n);
+    data->beta = (vx_float32 *)malloc(sizeof(vx_float32) * data->srcDescPtr->n);
```

Comment:
Do we need not allocate memory for ROI also here?

Reply:
Not required since we are passing ROI as vx_tensors now and directly we query from that